app/controllers/my_facilities/drug_stocks_controller.rb

```diff
@@ -92,6 +91,14 @@ def set_region
       end
   end
 
+  def blocks_to_display
+    if @selected_facility_sizes != @facility_sizes || @selected_zones != @zones
+      Region.none
+    else
+      @query.blocks.order(:name)
+    end
```

Comment:
We could flip this conditional, it would read a bit better, I think.

`if sizes same and zones same, blocks.order(stuff)`